smt/surrogate_models/ls.py

```diff
@@ -28,7 +28,11 @@ def _initialize(self):
         super(LS, self)._initialize()
         declare = self.options.declare
         supports = self.supports
-
+        declare(
```

Comment:
Forgot this one! Why did you add `xlimits` option to `LS` and `QP`? It does not seem to be used.

Reply:
In Mixed integer surrogate model, we have: 
        self._surrogate.options["xlimits"] = self._xlimits
Therefore, I put xlimits in every surrogate.

The question is: Why aren't the limits used to normalized data in LS & QP ? I haven't investigate LS or QP but I assume the data should also be normalized according to xlimits.

Reply:
Well... You add this line and I realize now that you introduce `xlimits` in `krg_based` to fix gower distance (I was fooled by this familiar name which is used by sampling methods). But so far surrogate models did not need `xlimits`! 

Moreover why just LS and QP and not the others IDW, RBF, etc. If we add it to all it has to be done in `SurrogateModel` class. But I do not think this is the right thing to do here. 

Gower distance is only used with kriging based models, and `xlimits` is only used with Gower distance. So for me it is a very specific option and should be named something like `gower_xlimits` to make it clear and `MixedIntegerSurrogateModel` should not set it (the surrogate should have been initialized properly with the right `gower_xlimits` when given to `MixedIntegerSurrogateModel`). Could you open a new PR for this ?

Reply:
Yes, I'll definitively check that out and open the PR when it's ready ! 

- I agree and I removed xlimits in QP & LS and have clean some stuff. 

- The mixed integer parameter "`xlimits`" is the same that is used for the sampling. It is also the same that we use for optimization and for the decorator.
- It makes sense to me that it keep its name. Typically, for optimization, the API do not ask to write `xlimits `in the parameters of both the optimizer, the decorator and the model. We could change all that for xtypes also but it would be another topic, I think.

- I also realized that the data were not normalized for QP (LS seems fine).